presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -0,0 +1,241 @@
+    private final ExtendedFileSystem fileSystem;
+
+    public AlluxioCachingFileSystem(Configuration configuration, URI uri, ExtendedFileSystem fileSystem)
+    {
+        this.fileSystem = fileSystem;
+    }
+
+    @Override
+    public synchronized void initialize(URI uri, Configuration configuration)
+            throws IOException
+    {
+        requireNonNull(uri, "uri is null");
+        requireNonNull(configuration, "configuration is null");
+        this.cachingFileSystem = new AlluxioCachingFileSystemInternal(configuration, uri, fileSystem);
+        cachingFileSystem.initialize(uri, configuration);
+    }
+
+    @Override
+    public String getScheme()
+    {
+        return cachingFileSystem.getScheme();
+    }
+
+    @Override
+    public URI getUri()
+    {
+        return cachingFileSystem.getUri();
+    }
+
+    @Override
+    public FSDataInputStream open(Path path, int bufferSize)
+            throws IOException
+    {
+        return cachingFileSystem.open(path, bufferSize);
+    }
+
+    @Override
+    public FSDataOutputStream create(Path path, FsPermission permission, boolean overwrite, int bufferSize, short replication, long blockSize, Progressable progress)
+            throws IOException
+    {
+        return cachingFileSystem.create(path, permission,
+                overwrite, bufferSize, replication, blockSize, progress);
+    }
+
+    @Override
+    public FSDataOutputStream append(Path path, int bufferSize, Progressable progress)
+            throws IOException
+    {
+        return cachingFileSystem.append(path, bufferSize, progress);
+    }
+
+    @Override
+    public boolean rename(Path source, Path destination)
+            throws IOException
+    {
+        return cachingFileSystem.rename(source, destination);
+    }
+
+    @Override
+    public boolean delete(Path path, boolean recursive)
+            throws IOException
+    {
+        return cachingFileSystem.delete(path, recursive);
+    }
+
+    @Override
+    public FileStatus[] listStatus(Path path)
+            throws IOException
+    {
+        return cachingFileSystem.listStatus(path);
+    }
+
+    @Override
+    public void setWorkingDirectory(Path workingDirectory)
+    {
+        cachingFileSystem.setWorkingDirectory(workingDirectory);
+    }
+
+    @Override
+    public Path getWorkingDirectory()
+    {
+        return cachingFileSystem.getWorkingDirectory();
+    }
+
+    @Override
+    public boolean mkdirs(Path path, FsPermission permission)
+            throws IOException
+    {
+        return cachingFileSystem.mkdirs(path, permission);
+    }
+
+    @Override
+    public FileStatus getFileStatus(Path path)
+            throws IOException
+    {
+        return cachingFileSystem.getFileStatus(path);
+    }
+
+    @Override
+    public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
+            throws Exception
+    {
+        if (hiveFileContext.isCacheable()) {
+            return cachingFileSystem.open(path);
```

Comment:
this is going to be an issue; we should use `openFile(path, hiveFileContext)` interface

Reply:
right. fixed